cypress/support/generateUserData.js

```diff
@@ -0,0 +1,81 @@
+const { faker } = require('@faker-js/faker');
+
+function generateData() {
+  const firstName = faker.person.firstName();
+  const lastName = faker.person.lastName();
+  const email = faker.internet.email({ firstName });
+  const mobileNumber = faker.phone.number('##########');
+
+  return { firstName, lastName, email, mobileNumber };
+}
+
+const getGenderAccordingToName = (firstName) => {
+  const maleNames = ['John', 'James', 'David', 'Michael', 'Robert', 'Jimmy'];
+  const femaleNames = ['Mary', 'Jennifer', 'Linda', 'Patricia', 'Jessica'];
+
+  if (maleNames.includes(firstName)) {
+    return 'male';
+  }
+
+  if (femaleNames.includes(firstName)) {
+    return 'female';
+  }
+
+  return faker.person.sex();
```

Comment:
The `faker.person.sex()` method may not always align with the names in your predefined lists. Consider adding more names to the `maleNames` and `femaleNames` arrays or using a different approach to ensure consistency.